### PR TITLE
TOK-405: claim builder rewards button

### PR DESCRIPTION
**Pull request by jurajpiar**
# What

Adds a button to claim builder rewards

[Task TOK-405](https://rsklabs.atlassian.net/browse/TOK-405)

# Testing

To test it I had to whitelist a new builder on v2 builder registry using:

```shell
cast send --private-key $PRIVATE_KEY --rpc-url $RPC_URL $SPONSOR_MANAGER_PROXY_ADDR "whitelistBuilder(address)" $ALICE --legacy

#and 

cast send --private-key $PRIVATE_KEY --rpc-url $RPC_URL $SPONSOR_MANAGER_PROXY_ADDR "activateBuilder(address,address,uint64)" $ALICE $ALICE 100000000000000000 --legacy
```

I have not tested that the SC does what it is supposed to do, as I could not allocate votes to them for some reason. The tx:

```shell
cast send --private-key $PRIVATE_KEY --rpc-url $RPC_URL $SPONSOR_MANAGER_PROXY_ADDR "allocate(address,uint256)" $ALICES_GAUGE 100000000000000000 --legacy
```

reverts without data/logs. Probably because of the lack of stRIF.

### CODE REVIEW

File: .env.testnet.qa [hunk 1]
```diff
@@ -18,6 +18,8 @@ NEXT_PUBLIC_CHAIN_ID=31
 
 # CR-related env variables
 NEXT_PUBLIC_SIMPLIFIED_REWARD_DISTRIBUTOR_ADDRESS=0xc469Cc2579De5C16210e9063B4E628bF8C46bA02
+NEXT_PUBLIC_SPONSORS_MANAGER_ADDRESS=
```
[the remark this answers is not in the document]
Comment (antomor) [reply .env.testnet.qa:21]: Furthermore, could we update the dev environment, please?

File: .env.testnet.local [hunk 2]
```diff
@@ -18,8 +18,12 @@ NEXT_PUBLIC_CHAIN_ID=31
 
 # CR-related env variables
 NEXT_PUBLIC_SIMPLIFIED_REWARD_DISTRIBUTOR_ADDRESS=0x4e84FCc953dE129C6C47c5B0AD7E57B226093Ae1
+NEXT_PUBLIC_SPONSORS_MANAGER_ADDRESS=0x0a01A9448883E5b83FbbCE9B103D7c4B1BB0E05b
+NEXT_PUBLIC_REWARD_DISTRIBUTOR_ADDRESS=0xcb2fdF1E05aC2885DA2fabBF0F37c47519C371a2
+
 NEXT_PUBLIC_EPOCH_DURATION_IN_DAYS=7
 NEXT_PUBLIC_FIRST_EPOCH_START_DATE_ISO="1970-01-01T00:00:00Z"
+NEXT_PUBLIC_WHITELISTED_BUILDERS_URL=http://localhost:3000/bim/whitelisted/
```
Comment (antomor) [inline .env.testnet.local:26]: I don't think we need this

File: src/app/collective-rewards/hooks/builderHooks.ts [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+
+    if (receiptError) {
+      return {
+        ...receiptError,
+        shortMessage: `Failed to get claim execution receipt (${receiptError.message})`,
+      }
+    }
+
+    if (isFetched && gauge === zeroAddress) {
+      return createZeroAddressError('Gauge', { builder })
```
Comment (antomor) [inline src/app/collective-rewards/hooks/builderHooks.ts:52]: Should we be more specific and say "The gauge doesn't exist"?

File: src/app/collective-rewards/hooks/useGetIsWhitelistedBuilder.ts [hunk 4]
```diff
@@ -22,3 +23,23 @@ export const useGetIsWhitelistedBuilder = (builder: Address) => {
...
+
+export const useGetIsWhitelistedBuilderV2 = (builder: Address) => {
+  const { data, isLoading, error } = useReadContract({
+    abi: BuilderRegistryAbi,
+    address: SponsorsManagerAddress,
+    functionName: 'isBuilderOperational',
+    args: [builder],
+    query: {
+      // 30 seconds, it's the mean time required for a block to be included in RSKJ
+      refetchInterval: 30_000,
```
Comment (antomor) [inline src/app/collective-rewards/hooks/useGetIsWhitelistedBuilder.ts:35]: Could we store it in a shared constant value?

Comment (jurajpiar) [reply src/app/collective-rewards/hooks/useGetIsWhitelistedBuilder.ts:35]: we could, but I'd rather not to that in this PR as that refactor is going to touch quite a lot of files.

File: src/app/collective-rewards/hooks/builderHooks.ts (hunk 3 above)
Comment (jurajpiar) [reply src/app/collective-rewards/hooks/builderHooks.ts:52]: I thought we could have a unified error for this, as we're going to check for non-zero address in other places, but sure. I can change it to a specific error

File: .env.testnet.local (hunk 2 above)
Comment (jurajpiar) [reply .env.testnet.local:26]: sure, though the .local file shouldn't matter much. in fact, now I'm thinking that maybe we should put it to .gitignore, wdyt?

File: .env.testnet.qa (hunk 1 above)
[the remark this answers is not in the document]
Comment (jurajpiar) [reply .env.testnet.qa:21]: could we have a chat about this pls

File: .env.testnet.local (hunk 2 above)
Comment (franciscotobar) [reply .env.testnet.local:26]: I have my own configuration :)

File: src/app/collective-rewards/hooks/builderHooks.ts (hunk 3 above)
Comment (jurajpiar) [reply src/app/collective-rewards/hooks/builderHooks.ts]: done

File: src/lib/contracts.ts [hunk 5]
```diff
@@ -44,7 +46,7 @@ const treasuryContracts = [
   { name: 'Grants', address: GRANTS_BUCKET_ADDRESS },
   { name: 'Grants - Active', address: GRANTS_ACTIVE_BUCKET_ADDRESS },
   { name: 'Growth', address: GROWTH_BUCKET_ADDRESS },
-  { name: 'Growth - Rewards', address: SIMPLIFIED_REWARD_DISTRIBUTOR_ADDRESS },
+  { name: 'Growth - Rewards', address: SPONSORS_MANAGER_ADDRESS },
```
[the remark this answers is not in the document]
Comment (jurajpiar) [reply src/lib/contracts.ts]: done

File: src/app/collective-rewards/hooks/useGetIsWhitelistedBuilder.ts (hunk 4 above)
Comment (jurajpiar) [reply src/app/collective-rewards/hooks/useGetIsWhitelistedBuilder.ts]: done

File: .env.testnet.local (hunk 2 above)
Comment (jurajpiar) [reply .env.testnet.local]: done

File: .env.testnet.qa (hunk 1 above)
[the remark this answers is not in the document]
Comment (jurajpiar) [reply .env.testnet.qa]: done

**Review by antomor — APPROVED** [review]
LGTM

**Pull request merged**